glean-core/src/metrics/labeled.rs

```diff
@@ -24,6 +28,46 @@ pub type LabeledBoolean = LabeledMetric<BooleanMetric>;
 /// A labeled string.
 pub type LabeledString = LabeledMetric<StringMetric>;
 
+/// A labeled custom_distribution.
+pub type LabeledCustomDistribution = LabeledMetric<CustomDistributionMetric>;
+
+/// A labeled memory_distribution.
+pub type LabeledMemoryDistribution = LabeledMetric<MemoryDistributionMetric>;
+
+/// A labeled timing_distribution.
+pub type LabeledTimingDistribution = LabeledMetric<TimingDistributionMetric>;
+
+/// The metric data needed to construct inner submetrics.
+///
+/// Different Labeled metrics require different amounts and kinds of information to
+/// be constructed.
+pub enum LabeledMetricData {
```

Comment:
It's unfortunate that this makes the memory size larger for the common case, but ... eh. it's gonna be fine.

Reply:
Ah, because it needs to store which variant it is for runtime type inspection?

Reply:
yeah, enums are always as big as the biggest variant (plus a tag somewhere so it knows which variant is active).